x-pack/plugins/cases/server/connectors/cases/cases_connector_executor.ts

```diff
@@ -742,35 +751,58 @@ export class CasesConnectorExecutor {
   private getCreateCaseRequest(
     params: CasesConnectorRunParams,
     groupingData: GroupedAlertsWithCaseId,
-    customFieldsConfigurations?: CustomFieldsConfiguration
+    customFieldsConfigurations?: CustomFieldsConfiguration,
+    templatesConfigurations?: TemplatesConfiguration
   ): Omit<BulkCreateCasesRequest['cases'][number], 'id'> & { id: string } {
     const { grouping, caseId, oracleRecord } = groupingData;
     const flattenGrouping = getFlattenedObject(grouping);
 
-    const requiredCustomFields = buildRequiredCustomFieldsForRequest(customFieldsConfigurations);
+    const selectedTemplate = templatesConfigurations?.find(
+      (template) => template.key === params.templateId
+    );
+    const caseFieldsFromTemplate = selectedTemplate ? selectedTemplate.caseFields : null;
+
+    const builtCustomFields = buildCustomFieldsForRequest(
+      customFieldsConfigurations,
+      caseFieldsFromTemplate?.customFields
+    );
+
     this.logger.debug(
-      `[CasesConnector][CasesConnectorExecutor][getCreateCaseRequest] Built ${requiredCustomFields.length} required custom fields for case with id ${caseId}`,
+      `[CasesConnector][CasesConnectorExecutor][getCreateCaseRequest] Built ${builtCustomFields.length} required custom fields for case with id ${caseId}`,
       this.getLogMetadata(params, {
         labels: {
           caseId,
-          totalCreatedCustomFields: requiredCustomFields.length,
+          totalCreatedCustomFields: builtCustomFields.length,
         },
         tags: ['case-connector:getCreateCaseRequest'],
       })
     );
 
     return {
       id: caseId,
-      description: this.getCaseDescription(params, flattenGrouping),
-      tags: this.getCaseTags(params, flattenGrouping),
-      title: this.getCasesTitle(params, flattenGrouping, oracleRecord.counter),
-      connector: { id: 'none', name: 'none', type: ConnectorTypes.none, fields: null },
+      description:
+        caseFieldsFromTemplate?.description ?? this.getCaseDescription(params, flattenGrouping),
+      tags: this.getCaseTags(params, flattenGrouping, caseFieldsFromTemplate?.tags),
+      title:
+        caseFieldsFromTemplate?.title ??
+        this.getCasesTitle(params, flattenGrouping, oracleRecord.counter),
+      connector: caseFieldsFromTemplate?.connector ?? {
+        id: 'none',
+        name: 'none',
+        type: ConnectorTypes.none,
+        fields: null,
+      },
       /**
        * TODO: Turn on for Security solution
        */
-      settings: { syncAlerts: false },
+      settings: caseFieldsFromTemplate?.settings ?? { syncAlerts: false },
+      ...(caseFieldsFromTemplate?.assignees
```

Comment:
[This test ](https://github.com/elastic/kibana/pull/190701/files#diff-63680614f782fb7e02ed92ca8447b40cee81a64684d6f893be0d4b94ada2142b)`creates new case with template`  adds all fields to templates and verify them in case in snapshot. 

Do you mean to verify each case field's value? Or maybe a functional test in a separate PR?